### ISSUE TITLE
Non PVP compliant Version Numbers

### ISSUE THREAD

**Issue by ffaf1** [haskell/cabal#8438]
**Describe the bug**
`cabal` does not complain, neither at parse time or with `cabal check`, about [incorrect PVP version numbers](https://pvp.haskell.org/#version-numbers).

**To Reproduce**

```
cabal get scroll
cd scroll-1.20180421/   # notice incorrect, 2 numbers only, version number
cabal check
```

**Expected behavior**
Something has to flag it as an error/warning, although I am not sure it is better at parse time or at check time.

**System information**
- Debian
- cabal version 3.6.2.0

**Comment by jneira** [comment]
So a warning should be a good fit for SHOULD 😉

**Comment by gbaz** [comment]
Agreed on a warning. If I recall we can handle them just fine in code, but the lexicographic ordering between 1.1 and 1.1.0 can be very confusing to people, so we want to discourage the former.
